cmd/kubeadm/app/util/endpoint_test.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"testing"
+)
+
+func TestGenerateMasterEndpoint(t *testing.T) {
+	var tests = []struct {
+		address  string
+		port     int32
+		endpoint string
+	}{
+		{
+			address:  "1.2.3.4",
+			port:     6443,
+			endpoint: "https://1.2.3.4:6443",
+		},
+		{
+			address:  "2001:db8::4",
+			port:     6443,
+			endpoint: "https://[2001:db8::4]:6443",
+		},
```

Comment:
Nit: could add a failure mode test case (invalid IP) to fully exercise all paths in GenerateMasterEndpoint, and check the error message.